### PR TITLE
Create overhaul

### PR DESCRIPTION
**Pull request by nibhar**
Removes export options and applies new design to the create flow.

Also updates PassphraseBox and PassphraseSetterBox. Passphrase renamed to password where applicable.

### CODE REVIEW

File: src/common.css [hunk 1]
```diff
@@ -57,6 +57,18 @@
     left: 4.5rem;
     top: 4.5rem;
     cursor: pointer;
+    transition: opacity .3s ease, transform .4s ease, -webkir-transform .4s ease;
```
Comment (sisou) [inline src/common.css]:
```suggestion
    transition: opacity .3s ease, transform .4s ease, -webkit-transform .4s ease;
```

File: src/common.css [hunk 2]
```diff
@@ -57,6 +57,18 @@
...
+}
+
+.page-header-back-button:hover {
+    opacity: .4;
+    transform: translate3D(-3px,0,0);
+}
+
+.page-header-back-button:focus {
+    opacity: .7; /* Ideally, we could change the color to --nimiq-light-blue on :focus instead, but this works for now */
+    transform: translate3D(-3px,0,0);
```
Comment (sisou) [inline src/common.css]: Missing spaces after commas.

File: src/components/PassphraseBox.css [hunk 3]
```diff
@@ -59,81 +64,98 @@
...
-    text-decoration: underline;
     cursor: pointer;
+    display: block;
+    font-size: 1.625rem;
+    transform: scale(1);
+    transition: transform .45s ease;
+}
+
+.passphrase-box .password-skip i.nq-icon {
+    background-image: url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" color="%23ffffff" viewBox="0 0 10 17"><polygon fill="currentColor" points="1.43,17 0,15.58 7.13,8.5 0,1.42 1.43,0 10,8.5 "/></svg>');
```
Comment (sisou) [inline src/components/PassphraseBox.css]: Can use `white` instead of `%23ffffff`.

File: src/components/PassphraseBox.css [hunk 4]
```diff
@@ -59,81 +64,98 @@
...
+    margin-left: 0.5rem;
+    height: 0.875rem;
+    width: 0.875rem;
+    transition: transform .3s ease;
+}
+
+.password-skip:hover,
+.password-skip:focus {
+  transform: scale(1.03);
```
Comment (sisou) [inline src/components/PassphraseBox.css]: I think we should remove scale transforms, especially those small ones, as they just look blurry.

File: src/components/PassphraseBox.css [hunk 5]
```diff
@@ -59,81 +64,98 @@
...
+  outline: none;
 }
 
-.passphrase-box.setter .submit {
-    margin-top: -9rem;
+.password-skip:hover i.nq-icon,
+.password-skip:focus i.nq-icon {
```
Comment (sisou) [inline src/components/PassphraseBox.css]: It's OK to not specify `i` here.

File: src/components/PassphraseSetterBox.js [hunk 6]
```diff
@@ -115,6 +117,7 @@ class PassphraseSetterBox extends Nimiq.Observable {
      * @param {Event} event
      */
     _onSubmit(event) {
+        console.log(event);
```
Comment (sisou) [inline src/components/PassphraseSetterBox.js]: Debugging statement can be removed.

File: src/request/create/Create.css [hunk 7]
```diff
@@ -1,63 +1,77 @@
...
 
-.page#recovery-words .skip {
-    color: inherit;
-    cursor: pointer;
+.wallet-identifier .identicon {
+    width: 7rem;
+    height: 6rem;
+    position: absolute;
+    transform: translate(0.2rem, 3.5rem) scale(0);
+    animation: pop-in-13 0.4s cubic-bezier(.73,2.08,.48,1);
```
Comment (sisou) [inline src/request/create/Create.css]: What does the `-13` in the animation name stand for?

File: src/request/create/Create.css [hunk 8]
```diff
@@ -1,63 +1,77 @@
...
 }
 
-.page-header .progress-indicator {
+.wallet-identifier .blank {
+    width: 7rem;
+    height: 6rem;
     position: absolute;
-    left: 0;
-    top: 0;
+    background-image: url("data:image/svg+xml,%3Csvg width='58' height='49' viewBox='0 0 58 49' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath opacity='0.08' d='M56.8082 22.2407L44.9478 2.67489C44.1017 1.27518 42.5283 0.417297 40.8362 0.417297H17.1154C15.4233 0.417297 13.8578 1.28271 13.0117 2.67489L1.14344 22.2407C0.297401 23.6404 0.297401 25.3562 1.14344 26.7559L13.0038 46.3217C13.8499 47.7214 15.4233 48.5792 17.1154 48.5792H40.8362C42.5283 48.5792 44.0938 47.7138 44.9399 46.3217L56.8002 26.7559C57.6542 25.3637 57.6542 23.6404 56.8082 22.2407Z' fill='%231F2348'/%3E%3C/svg%3E%0A");
```
Comment (sisou) [inline src/request/create/Create.css]: I don't think it's necessary to escape triangle-brackets.

File: src/request/create/Create.css [hunk 9]
```diff
@@ -1,63 +1,77 @@
...
 
-.page-header .page-header-back-button {
-    top: 5.5rem;
-}
+.wallet-identifier .blank:nth-child(1) { transform: translate(6.5rem, 0rem); }
+.wallet-identifier .blank:nth-child(2) { transform: translate(13rem, 3.5rem); }
+.wallet-identifier .blank:nth-child(3) { transform: translate(13rem, 10.5rem); }
+.wallet-identifier .blank:nth-child(4) { transform: translate(6.5rem, 14rem); }
+.wallet-identifier .blank:nth-child(5) { transform: translate(0rem, 10.5rem); }
+.wallet-identifier .identicon { transform: translate(0.2rem, 3.5rem) scale(1.15); } /* TODO: update as soon as new identicons hit */
```
Comment (sisou) [inline src/request/create/Create.css]: I think it's ok to remove already in preparation for Identicon change.

File: src/request/create/IdenticonSelector.css [hunk 10]
```diff
@@ -52,13 +54,27 @@
...
+.identicon-selector .wrapper:nth-child(4):hover { transform: translate(  0.0rem,   0.00rem) scale(1.1); }
+.identicon-selector .wrapper:nth-child(5) { transform: translate(-12.5rem,   7.25rem) scale(1); }
+.identicon-selector .wrapper:nth-child(5):focus,
+.identicon-selector .wrapper:nth-child(5):hover { transform: translate(-12.5rem,   7.25rem) scale(1.1); }
+.identicon-selector .wrapper:nth-child(6) { transform: translate( 12.5rem,   7.25rem) scale(1); }
+.identicon-selector .wrapper:nth-child(6):focus,
+.identicon-selector .wrapper:nth-child(6):hover { transform: translate( 12.5rem,   7.25rem) scale(1.1); }
+.identicon-selector .wrapper:nth-child(7) { transform: translate(  0.0rem,  14.25rem) scale(1); }
+.identicon-selector .wrapper:nth-child(7):focus,
+.identicon-selector .wrapper:nth-child(7):hover { transform: translate(  0.0rem,  14.25rem) scale(1.1); }
```
Comment (sisou) [inline src/request/create/IdenticonSelector.css]: I think what you can do here to make this look nicer and remove code duplication, is translate-transform the `.wrapper`, but scale-transfrom the `.wrapper > .identicon`.

Comment (danimoh) [reply src/request/create/IdenticonSelector.css]: Should be possible. For this, you'd need to set the default identicon scale to 1 and the `pop-in` `animation-direction` to `backwards`.

File: src/request/create/IdenticonSelector.js [hunk 11]
```diff
@@ -114,42 +101,23 @@ class IdenticonSelector extends Nimiq.Observable {
...
-
-    /**
-     * @private
-     */
-    _onSelectionConfirmed() {
-        if (!this._selectedAddress) { // something went wrong
-            this._clearSelection(); // clear current selection
+    _onSelectionConfirmed(selectedAddress, e) {
+        e.preventDefault();
+        if (!selectedAddress) { // something went wrong
```
Comment (danimoh) [inline src/request/create/IdenticonSelector.js]: This should never happen and this code block can be removed?

File: src/request/create/index.html [hunk 12]
```diff
@@ -82,104 +70,40 @@
             <div class="page-header nq-card-header">
                 <div class="progress-indicator"></div>
                 <a tabindex="0" class="page-header-back-button nq-icon arrow-left"></a>
-                <h1 data-i18n="create-heading-choose-identicon" class="nq-h1">Choose your avatar</h1>
+                <h1 data-i18n="create-heading-choose-identicon" class="nq-h1">Choose an avatar</h1>
```
Comment (danimoh) [inline src/request/create/index.html]: Capitalize avatar?

File: src/request/create/index.html [hunk 13]
```diff
@@ -82,104 +70,40 @@
...
             </div>
         </div>
 
         <div id="set-passphrase" class="page nq-card">
             <div class="page-header nq-card-header">
                 <div class="progress-indicator"></div>
                 <a tabindex="0" class="page-header-back-button nq-icon arrow-left"></a>
-                <h1 data-i18n="create-heading-keyfile" class="nq-h1">This is your Key File</h1>
+                <h1 class="nq-h1 hide-on-repetition" data-i18n="create-heading-create-password">Create a password</h1>
+                <h1 class="nq-h1 show-on-repetition" data-i18n="create-heading-repeat-password">Confirm your password</h1>
```
Comment (danimoh) [inline src/request/create/index.html]: Capitalize password?

File: src/request/create/index.html [hunk 14]
```diff
@@ -82,104 +70,40 @@
...
             </div>
 
             <div class="page-body nq-card-body">
-                <div class="keyfile-info">
-                    <div class="nq-icon walletfile"></div>
-                    <div class="keyfile-text">
-                        <p data-i18n="create-text-keyfile-info" class="nq-text">Your Key File gives you full access to your wallet. You'll need it everytime you log in.</p>
-                        <p data-i18n="create-hint-keyfile-password" class="nq-text">To protect your wallet, first protect it with a password.</p>
-                    </div>
+                <div class="wallet-identifier inactive">
```
Comment (danimoh) [inline src/request/create/index.html]: `inactive` seems to be unused and can be removed

File: src/translations/index.js [hunk 15]
```diff
@@ -22,15 +22,11 @@ const TRANSLATIONS = {
...
-        'create-text-select-avatar': 'Select an avatar for your wallet\'s default account from the selection below.',
-        'create-hint-more-accounts': 'You can add more accounts later.',
-        'create-heading-keyfile': 'This is your Key File',
-        'create-text-keyfile-info': 'Your Key File gives you full access to your wallet. '
-                                  + 'You\'ll need it everytime you log in.',
-        'create-hint-keyfile-password': 'To protect your wallet, first protect it with a password.',
-        'create-heading-backup-account': 'Create a backup',
+        'create-heading-choose-identicon': 'Choose an avatar',
+        'create-heading-create-password': 'Create a password',
+        'create-heading-repeat-password': 'Confirm your password',
```
Comment (danimoh) [inline src/translations/index.js]: Mh, ok, what is our standard for capitalization of headings?

File: src/translations/index.js [hunk 16]
```diff
@@ -152,14 +149,10 @@ const TRANSLATIONS = {
...
-        'create-hint-more-accounts': 'Neue Konten kannst du später hinzufügen.',
-        'create-heading-keyfile': 'Das ist deine Wallet Datei',
-        'create-text-keyfile-info': 'Deine Wallet Datei gibt dir vollen Zugang zu deiner Wallet. '
-                                  + 'Du brauchst sie jedesmal wenn du dich einloggst.',
-        'create-hint-keyfile-password': 'Um deine Wallet zu schützen, schütze es mit einem Passwort.',
-        'create-heading-backup-account': 'Erstelle ein Backup',
+        'create-heading-create-password': 'Erstelle ein Passwort',
+        'create-heading-repeat-password': 'Bestätige dein Passwort',
         'create-heading-validate-backup': 'Überprüfe dein Backup',
+        'create-wallet-desc': 'Das ist dein Wallet mit deinem ersten Account.',
```
Comment (danimoh) [inline src/translations/index.js]: `Das ist dein Account mit deiner ersten Addresse.`

File: src/components/PassphraseBox.css [hunk 17]
```diff
@@ -59,81 +64,98 @@
...
-    opacity: 0;
-}
-
 .passphrase-box.hide-cancel .cancel {
     display:none;
 }
 
-/* S E T T E R */
+.passphrase-box .prompt {
+    margin-top: -2.75rem;
```
[the remark this answers is not in the document]
Comment (nibhar) [reply src/components/PassphraseBox.css]: Noted down, will take care of it in the checkout overhaul PR, as there is no close button here.

File: src/components/PassphraseBox.css [hunk 18]
```diff
@@ -59,81 +64,98 @@
...
+    top: 1.5rem;
+    left: 2rem;
+    opacity: .5;
+    transition: opacity .3s ease, transform .45s ease;
+}
+
+.passphrase-input .eye-button:hover,
+.passphrase-input .eye-button:focus {
+    opacity: 1;
+    transform: scale(1.03);
```
[the remark this answers is not in the document]
Comment (nibhar) [reply src/components/PassphraseBox.css]: It really does not look good that way. I decided, to remove the small scaling factor and do the transition on opacity only for now.

File: src/request/create/Create.js [hunk 19]
```diff
@@ -21,114 +17,74 @@ class Create {
...
-        this._downloadKeyfile = new DownloadKeyfile($downloadKeyfile);
-        this._privacyAgent = new PrivacyAgent($privacyAgent);
-        this._recoveryWords = new RecoveryWords($recoveryWords, false);
-        this._validateWords = new ValidateWords($validateWords);
+        this._identiconSelector = new IdenticonSelector(this.$identiconSelector, request.defaultKeyPath);
         this._passphraseSetter = new PassphraseSetterBox($setPassphrase);
+        // Set up progress indicators
+        /* eslint-disable-next-line no-new */
+        new ProgressIndicator(document.querySelector(`#${Create.Pages.CHOOSE_IDENTICON} .progress-indicator`), 3, 1);
+        this.progressIndicator = new ProgressIndicator(
```
[the remark this answers is not in the document]
Comment (nibhar) [reply src/request/create/Create.js]: one for each page. They create markup so they are needed for each page individually.

File: src/translations/index.js (hunk 15 above)
Comment (nibhar) [reply src/translations/index.js]: a very good question. I assumed we would not capitalize, as none of the headlines had capitalization before except for `Key File` which we capitalize everywhere. How about doing this (or don't, depending on decision) in another PR once wording is finalized?

File: src/request/create/Create.css (hunk 7 above)
Comment (nibhar) [reply src/request/create/Create.css]: renamed to `pop-in-0-to-115`

File: src/request/create/IdenticonSelector.css (hunk 10 above)
Comment (nibhar) [reply src/request/create/IdenticonSelector.css]: `.wrapper` translated to correct position
`.wrapper .identicon` for the pop-in scale
`.wrapper .identicon img` for `:hover` and `:active` scale

File: src/request/create/index.html [hunk 20]
```diff
@@ -82,104 +70,40 @@
...
-                        <p data-i18n="create-text-keyfile-info" class="nq-text">Your Key File gives you full access to your wallet. You'll need it everytime you log in.</p>
-                        <p data-i18n="create-hint-keyfile-password" class="nq-text">To protect your wallet, first protect it with a password.</p>
-                    </div>
+                <div class="wallet-identifier">
+                    <div class="blank"></div>
+                    <div class="blank"></div>
+                    <div class="blank"></div>
+                    <div class="blank"></div>
+                    <div class="blank"></div>
+                    <div class="identicon"></div>
```
Comment (sisou) [inline src/request/create/index.html]: The wallet identifier should become a component as soon as we need it somewhere else.

File: src/request/create/Create.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (danimoh) [reply src/request/create/Create.css]: Resolved but not integrated?

File: src/components/PassphraseSetterBox.js [hunk 21]
```diff
@@ -95,18 +93,22 @@ class PassphraseSetterBox extends Nimiq.Observable {
...
         await AnimationUtils.animate('shake', $hint);
     }
 
     /**
      * @param {boolean} isValid
      */
     _onInputChangeValidity(isValid) {
+        if (this._password && this._passphraseInput.text === this._password) {
+            this.fire(PassphraseSetterBox.Events.SUBMIT, this._password);
+        }
```
Comment (sisou) [inline src/components/PassphraseSetterBox.js]: Is this supposed to continue executing or does it need a `return` statement?

File: src/request/create/Create.css [hunk 22]
```diff
@@ -1,63 +1,74 @@
...
 }
 
-.page-header .progress-indicator {
+.wallet-identifier .blank {
+    width: 7rem;
+    height: 6rem;
     position: absolute;
-    left: 0;
-    top: 0;
+    background-image: url("data:image/svg+xml,<svg width='58' height='49' viewBox='0 0 58 49' fill='none' xmlns='http://www.w3.org/2000/svg'><path opacity='0.08' d='M56.8082 22.2407L44.9478 2.67489C44.1017 1.27518 42.5283 0.417297 40.8362 0.417297H17.1154C15.4233 0.417297 13.8578 1.28271 13.0117 2.67489L1.14344 22.2407C0.297401 23.6404 0.297401 25.3562 1.14344 26.7559L13.0038 46.3217C13.8499 47.7214 15.4233 48.5792 17.1154 48.5792H40.8362C42.5283 48.5792 44.0938 47.7138 44.9399 46.3217L56.8002 26.7559C57.6542 25.3637 57.6542 23.6404 56.8082 22.2407Z' fill='%231F2348'/></svg>");
```
Comment (sisou) [inline src/request/create/Create.css]: In the dataurl above, you are using single quotes for the string, and double quotes for the svg attributes. This is the other way around. I like the first approach better.

File: src/request/create/Create.js [hunk 23]
```diff
@@ -21,114 +17,74 @@ class Create {
...
-                this._validateWords.setWords(mnemonic);
                 window.location.hash = Create.Pages.SET_PASSPHRASE;
+                // eslint-disable-next-line no-new
+                new Identicon(
+                    address,
+                    /** @type {HTMLDivElement} */(this.$walletIdentifier.querySelector('.identicon')),
+                );
+                this.progressIndicator.setStep(2);
                 this._passphraseSetter.reset();
                 this._passphraseSetter.focus();
```
Comment (sisou) [inline src/request/create/Create.js]: I think this `.focus()` also needs the logic about no autofocus on mobile.

Comment (nibhar) [reply src/request/create/Create.js]: It does, as do lots of other `.focus()` calls. I don't think maintaining that on many branches helps. I will open a new PR once the constant is merged.

File: src/request/create/Create.css (hunk 22 above)
Comment (nibhar) [reply src/request/create/Create.css]: I agree.

File: src/request/create/index.html (hunk 20 above)
Comment (danimoh) [reply src/request/create/index.html]: FYI: in the accounts manager it is already a component (also scales well and can display more than one identicon) which I'll eventually move to vue components.

File: src/components/PassphraseBox.css [hunk 24]
```diff
@@ -109,27 +109,26 @@
...
+    outline: none;
+    opacity: 0.7;
 }
 
-.password-skip:hover i.nq-icon,
-.password-skip:focus i.nq-icon {
-  transform: translate3D(3px, 0, 0);
+.passphrase-box .password-skip:hover .nq-icon,
+.passphrase-box .password-skip:focus .nq-icon {
+    transform: translate3D(-0.25rem, 0, 0);
```
Comment (danimoh) [inline src/components/PassphraseBox.css]: preferred the movement in the other direction.
Also I _might_ prefer to increase the opacity on hover instead the other way around.
But if you prefer it this way, ok.

**Pull request merged**